src/lib/features/feature-toggle/feature-toggle-strategies-store.ts

```diff
@@ -541,207 +541,209 @@ class FeatureStrategiesStore implements IFeatureStrategiesStore {
 
-        const sortByMapping = {
-            name: 'feature_name',
-            type: 'type',
-            lastSeenAt: 'env_last_seen_at',
-        };
+                if (userId) {
+                    query.leftJoin(`favorite_features`, function () {
+                        this.on(
+                            'favorite_features.feature',
+                            'features.name',
+                        ).andOnVal('favorite_features.user_id', '=', userId);
+                    });
+                    selectColumns = [
+                        ...selectColumns,
+                        this.db.raw(
+                            'favorite_features.feature is not null as favorite',
+                        ),
+                    ];
+                }
 
-        if (favoritesFirst) {
-            query = query.orderBy('favorite', 'desc');
-        }
+                selectColumns = [
+                    ...selectColumns,
+                    this.db.raw(
+                        'EXISTS (SELECT 1 FROM feature_strategies WHERE feature_strategies.feature_name = features.name AND feature_strategies.environment = feature_environments.environment) as has_strategies',
+                    ),
+                    this.db.raw(
+                        'EXISTS (SELECT 1 FROM feature_strategies WHERE feature_strategies.feature_name = features.name AND feature_strategies.environment = feature_environments.environment AND (feature_strategies.disabled IS NULL OR feature_strategies.disabled = false)) as has_enabled_strategies',
+                    ),
+                ];
+
+                const sortByMapping = {
+                    name: 'features.name',
+                    type: 'features.type',
+                    lastSeenAt: lastSeenQuery,
+                };
 
-        if (sortBy.startsWith('environment:')) {
-            const [, envName] = sortBy.split(':');
-            query = query
-                .orderByRaw(
-                    `CASE WHEN feature_environments.environment = ? THEN feature_environments.enabled ELSE NULL END ${validatedSortOrder} NULLS LAST`,
-                    [envName],
-                )
-                .orderBy('created_at', 'asc');
-        } else if (sortByMapping[sortBy]) {
-            query = query
-                .orderBy(sortByMapping[sortBy], validatedSortOrder)
-                .orderBy('created_at', 'asc');
-        } else {
-            query = query.orderBy('created_at', validatedSortOrder);
-        }
+                let rankingSql = 'order by ';
+                if (favoritesFirst) {
+                    rankingSql +=
+                        'favorite_features.feature is not null desc, ';
+                }
 
-        const total = await countQuery
-            .countDistinct({ total: 'features.name' })
-            .first();
+                if (sortBy.startsWith('environment:')) {
+                    const [, envName] = sortBy.split(':');
+                    rankingSql += this.db
```

Comment:
This is the existing sorting part, which we building into `denseRank` query. Knex does not support this function properly, so had to build raw query.